packages/web3-eth/src/eth_tx.ts

```diff
@@ -279,17 +263,23 @@ export const prepareTransactionForSigning = async (
 	web3Context: Web3Context<EthExecutionAPI>,
 	privateKey?: HexString | Buffer,
 ) => {
-	const populatedTransaction = await populateTransaction(
-		transaction,
+	const transactionBuilder = web3Context.transactionBuilder ?? defaultTransactionBuilder;
```

Comment:
I think top level function `prepareTransactionForSigning` should not be overriden instead inner parts. So why I kept `prepareTransactionForSigning` as it is and made `transactionBuilder` as override.

Reply:
We should remove the `TODO` on line `260`